receiver/dockerstatsreceiver/receiver.go

```diff
@@ -264,3 +265,9 @@ func (r *receiver) recordCPUMetrics(now pcommon.Timestamp, cpuStats *dtypes.CPUS
 		r.mb.RecordContainerCPUUsagePercpuDataPoint(now, int64(v), fmt.Sprintf("cpu%s", strconv.Itoa(coreNum)))
 	}
 }
+
+func (r *receiver) recordInspectMetrics(now pcommon.Timestamp, containerJSON *dtypes.ContainerJSON) {
```

Comment:
Why `inspect`? is it because these metrics can be gathered via `docker inspect`? If so, I'm not sure if it's a good name for it here, since we aren't using `docker inspect`, we are making API calls directly to the engine.

Perhaps it could be `recordHostConfigMetrics` and you can use the HostConfig type and move the restarts recording out into the calling function.

Reply:
Hi @jamesmoessis 
We named it docker inspect following the same naming docker api uses. For those metrics they call it "container inspect" endpoint (https://docs.docker.com/engine/api/v1.42/#tag/Container/operation/ContainerInspect)
But I do agree that the naming is not the best and I do prefer your suggestion of using HostConfig and the restarts, as it is isolated for now, call it directly from the caller function.
Will work on those changes.